src/game/DBScripts/ScriptMgr.cpp

```diff
@@ -2616,8 +2703,164 @@ bool ScriptAction::ExecuteDbscriptCommand(WorldObject* pSource, WorldObject* pTa
+                break;
+            }
+            case 3: // add creature to the formation
+            {
+                if (LogIfNotCreature(pSource))
+                    return false;
+
+                if (!formationData)
+                {
+                    sLog.outErrorDb(" DB-SCRIPTS: Process table `%s` id %u, command %u failed. %s is not in formation!",
+                        m_table, m_script->id, m_script->command, leader->GetGuidStr().c_str());
+                    break;
+                }
+                formationData->Add(static_cast<Creature*>(pSource));
+                break;
+            }
+            case 4: // remove creature from the formation
+            {
+                if (LogIfNotCreature(pSource))
+                    return false;
+
+                if (!formationData)
+                {
+                    sLog.outErrorDb(" DB-SCRIPTS: Process table `%s` id %u, command %u failed. %s is not in formation!",
+                        m_table, m_script->id, m_script->command, leader->GetGuidStr().c_str());
+                    break;
+                }
+                formationData->Remove(static_cast<Creature*>(pSource));
+                break;
+            }
+            case 5: // switch formation shape
+            {
+                if (!formationData)
+                {
+                    sLog.outErrorDb(" DB-SCRIPTS: Process table `%s` id %u, command %u failed. %s is not in formation!",
+                        m_table, m_script->id, m_script->command, leader->GetGuidStr().c_str());
+                    break;
+                }
+
+                if (m_script->formationData.data1 < static_cast<uint32>(SpawnGroupFormationType::SPAWN_GROUP_FORMATION_TYPE_COUNT))
+                {
+                    if (!formationData->SwitchFormation(static_cast<SpawnGroupFormationType>(m_script->formationData.data1)))
+                        sLog.outErrorDb(" DB-SCRIPTS: Process table `%s` id %u, command %u subcommand changeShape(%u) failed.",
+                            m_table, m_script->id, m_script->command, m_script->formationData.command);
+                }
+                else
+                {
+                    sLog.outErrorDb(" DB-SCRIPTS: Process table `%s` id %u, command %u subcommand switchShape(%u) failed, wrong shape id(%u).",
+                        m_table, m_script->id, m_script->command, m_script->formationData.command, m_script->formationData.data1);
+                    return true;
+                }
+                break;
+            }
+            case 6:  // set formation spread
+            {
+                if (!formationData)
+                {
+                    sLog.outErrorDb(" DB-SCRIPTS: Process table `%s` id %u, command %u failed. %s is not in formation!",
+                        m_table, m_script->id, m_script->command, leader->GetGuidStr().c_str());
+                    break;
+                }
+
+                if (m_script->x <= 15)
```

Comment:
This is a super odd way of using X. Use a different parameter. Do you really need a float?

Reply:
yes i can make it x10 maybe in some data if need but will be even less intuitive